Added Pantheon Online Accounts documentation.

tintou opened the pull request.

lewisgoddard commented:
Looks good web-wise, I have no idea about the actual document though.

danrabbit commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
+
+## Create a Provider {#poa-create-provider}
+
+There are several default providers shipped with elementary, and much more available in the AppCenter. If the provider you want is already available, jump to the next part as you *MUST* use it. Otherwise, here is how to create a new one.
```
Link on "jump to the next part"?

danrabbit commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
+
+## Create a Service {#poa-create-service}
+
+As in the Create a Provider section, there are several default services shipped with elementary, and much more available in the AppCenter. If the service you want is already available, jump to the next part as you *MUST* use it. Otherwise, here is how to create a new one.
```
Link "Create a Provider"

danrabbit replied:
Link "jump to the next part"

danrabbit commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
+|  `oauth2`  |       `web_server` or `user_agent`       |
+| `password` |                `password`                |
+
+To create a provider, you have to ship a .provider file in "/usr/share/accounts/providers/", it is better to name the file as `%provider.provider` (here it’s `google.provider` for example)
```
I think this needs more explanation. Where to I ship this file with my app? In the "data" directory? How do I install it to /usr/share/accounts/providers? What do I need to do to my CMake or with packaging?

danrabbit commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
+## Create a Service {#poa-create-service}
+
+As in the Create a Provider section, there are several default services shipped with elementary, and much more available in the AppCenter. If the service you want is already available, jump to the next part as you *MUST* use it. Otherwise, here is how to create a new one.
+The .service file should be stored in "/usr/share/accounts/services/", it is better to name the file as `%provider-%service.service` (here it’s `google-contacts.service` for example)
```
Same question as the above file. Where do I store this in my project? How do I install it with CMake? Do I have to do anything with packaging?

danrabbit commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
+        </application>
+
+In some cases, your application will handle all kind of services (such as e-mails). If so, you can just omit the `services` and just fill the `service-types` you want.
+
```
This seems really incomplete. I've created these files, now what? How does my app actually interact with the data that these things provide?

tintou commented:
Okay, I fixed the missing links and added the addition of the CMakeLists.txt
But you're right, I also need to merge the integration into an application
I already have a Google Document that I have to adapt in order to have the same amount of description as the rest of the documentation

nathandyer commented on docs/code/getting-started.md:
```diff
@@ -675,4 +675,162 @@ Let’s recap what we learned in this section:
+                            <setting name="AuthPath">/o/oauth2/auth</setting>
+                            <setting name="TokenPath">/o/oauth2/token</setting>
+                            <setting type="int" name="UiPolicy">0</setting>
+                            <setting name="RedirectUri">http://elementaryos.org</setting>
```
Here the RedirectUri is listed as "elementaryos.org," yet above it is "elementary.io." I imagine that we want to be consistent. Also, it's probably worth mentioning to a new developer that the URI you choose isn't all that important, but that it MUST match the redirect URI you provide to the actual service provider (Twitter, Facebook, etc.) when you sign up to make an application (unless I'm wrong about that part).

nathandyer commented:
What's there is pretty good. In case you all weren't aware/forgot, [here](https://docs.google.com/document/d/16l4pvhJdcL0V37JKojkyC3z8gGsFVLATDmoxPsSyR7Q/edit?usp=sharing) is a link to a Google Doc with documentation that someone has already written about how to retrieve the token from POA in a typical Vala app. It's even based around Google Contacts, so it should be easy to copy/paste from it.

danrabbit commented:
Closing because this hasn't moved in over a year
